src/base/bittorrent/torrenthandle.cpp

```diff
@@ -729,30 +733,7 @@ bool TorrentHandle::isSequentialDownload() const
 
 bool TorrentHandle::hasFirstLastPiecePriority() const
 {
-    if (!hasMetadata()) return false;
-
-    // Get int first media file
-    std::vector<int> fp;
-    fp = m_nativeHandle.file_priorities();
-
-    TorrentInfo::PieceRange extremities;
-    bool found = false;
-    int count = static_cast<int>(fp.size());
-    for (int i = 0; i < count; ++i) {
-        const QString ext = Utils::Fs::fileExtension(filePath(i));
-        if (Utils::Misc::isPreviewable(ext) && (fp[i] > 0)) {
-            extremities = info().filePieces(i);
-            found = true;
-            break;
-        }
-    }
-
-    if (!found) return false; // No media file
-
-    int first = m_nativeHandle.piece_priority(extremities.first());
-    int last = m_nativeHandle.piece_priority(extremities.last());
-
-    return ((first == 7) && (last == 7));
+    return m_hasFirstLastPiecePriority;
```

Comment:
>but I was asked to change it to m_hasFirstLastPiecePriority. That means that new flag is the canonical indicator as to whether a torrent has this option set, and it's no longer valid to check the actual priority of the pieces because it could be set before the torrent's metadata is available.

You was asked to change field name only.
>I think my earlier approach of using m_needsToSetFirstLastPiecePriority was better.

Let it be. Otherwise this changes will never be released.
@evsh? IMO, we should omit changing internals here.